test/tasks/ods-build-python_test.go

```diff
@@ -0,0 +1,73 @@
+package tasks
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/opendevstack/pipeline/pkg/tasktesting"
+)
+
+func TestTaskODSBuildPython(t *testing.T) {
+	runTaskTestCases(t,
+		"ods-build-python",
+		map[string]tasktesting.TestCase{
+			"task should build python flask app": {
+				WorkspaceDirMapping: map[string]string{"source": "python-flask-sample-app"},
+				PreRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+					ctxt.ODS = tasktesting.SetupGitRepo(t, ctxt.Namespace, wsDir)
+					ctxt.Params = map[string]string{
+						// configure the following lines if you need to run the test within your corporate network
+						// this is needed to allow pip to download pkgs
+						// "no-proxy":    "",
+						// "https-proxy": "",
+					}
+				},
+				WantRunSuccess: true,
+				PostRunFunc: func(t *testing.T, ctxt *tasktesting.TaskRunContext) {
+					wsDir := ctxt.Workspaces["source"]
+
+					wantFiles := []string{
+						"docker/app/main.py",
+						"docker/app/requirements.txt",
+						"build/test-results/test/report.xml",
+						"build/test-results/coverage/coverage.xml",
+						".ods/artifacts/xunit-reports/report.xml",
+						".ods/artifacts/code-coverage/coverage.xml",
+						".ods/artifacts/sonarqube-analysis/analysis-report.md",
+						".ods/artifacts/sonarqube-analysis/issues-report.csv",
+					}
+					for _, wf := range wantFiles {
+						if _, err := os.Stat(filepath.Join(wsDir, wf)); os.IsNotExist(err) {
+							t.Fatalf("Want %s, but got nothing", wf)
+						}
+					}
+
+					wantContains := `
```

Comment:
maybe load this from file? a typical convention is to use a folder named `golden` under `test/testdata` to place this.